src/main.ts

```diff
@@ -232,83 +230,234 @@ class AutoEntities extends LitElement {
-        const filters = await get_filter(this.hass, filter);
-        return filters;
-      })
-    );
+        const getEntAreaName = ent => {
+          let area;
+          if (ent.area_id) {
+            area = findArea(ent);
+          }
+          if (!area && ent.device_id !== undefined) {
+            const device = findDev(ent);
+            if (device) {
+              area = findArea(device);
+            }
+          }
+          if (area && area.name) {
+            return area.name;
+          }
+          return "";
+        };
 
-    const all_entities: EntityList = Object.keys(this.hass.states).map(format);
-    // Include
-    entities = entities.concat(
-      ...(await Promise.all(include_filters.map((f) => f(all_entities))))
-    );
-    // Exclude
-    entities = entities.filter((e) => !exclude_filters.some((f) => f(e)));
+        const getEntGroupName = group => {
+          const groupEnt = this.hass.states[group];
+          if (groupEnt && groupEnt.attributes.friendly_name) {
+            return groupEnt.attributes.friendly_name;
+          }
+          return "";
+        };
+
+        const getEntDeviceManufacturerName = ent => {
+          if (ent.device_id !== undefined) {
+            const device = findDev(ent);
+            if (device && device.manufacturer) {
+              return device.manufacturer;
+            }
+          }
+          return "";
+        };
 
-    // Global sort
-    const sorter = this._config.sort?.method
-      ? await get_sorter(this.hass, this._config.sort)
-      : (x) => x;
-    entities = await sorter(entities);
+        const getEntDeviceModelName = ent => {
+          if (ent.device_id !== undefined) {
+            const device = findDev(ent);
+            if (device && device.model) {
+              return device.model;
+            }
+          }
+          return "";
+        };
 
-    // Unique
-    if (this._config.unique) {
-      let sorter = (
-        entity: LovelaceRowConfig,
-        index: number,
-        self: LovelaceRowConfig[]
-      ) => index === self.findIndex((e) => compare_deep(e, entity));
-
-      if (this._config.unique === "entity") {
-        sorter = (
-          entity: LovelaceRowConfig,
-          index: number,
-          self: LovelaceRowConfig[]
-        ) => index === self.findIndex((e) => e.entity === entity.entity);
+        const getEntNameForConfig = (ent: any, config: EntityNameConfig) => {
+          const { text, area, group, device_manufacturer, device_model } = config;
+          if (text) {
+            return text;
+          } else if (area) {
+            return getEntAreaName(ent);
+          } else if (group) {
+            return getEntGroupName(group);
+          } else if (device_manufacturer) {
+            return getEntDeviceManufacturerName(ent);
+          } else if (device_model) {
+            return getEntDeviceModelName(ent);
+          }
+        };
+
+        const newEntities = entities.map(e => ({...e}));
```

Comment:
Suggest to just use regex replacement which can cover all these cases and then is extensible to cover replacements users may dream up.